packages/amplify_datastore/example/integration_test/relationship_type_test.dart

```diff
@@ -110,8 +110,9 @@ void main() {
     });
 
     group('BelongsTo', () {
-      var child = ChildModel(name: 'child');
-      var parent = BelongsToModel(name: 'HasOne', child: child);
+      var relationshipParent = ChildModel(name: 'child');
```

Comment:
If we want to make these tests easier to understand, I think we need to update the schema. `ChildModel` was probably a bad name choice (by me 😅 ) to begin with. `ChildModel` is currently related to `BelongsToModel` and `HasOneModel` in the schema. It might make sense to replace `ChildModel` with two separate models, like below. The schema below is using the v2 directives, but that could be modified.

```graphql
type HasOneModel @model @auth(rules: [{allow: public}]) {
  id: ID!
  name: String!
  childID: ID!
  child: HasOneChildModel @hasOne(fields: ["childID"])
}

type HasOneChildModel @model @auth(rules: [{allow: public}]) {
  id: ID!
  name: String!
}

type BelongsToModel @model @auth(rules: [{allow: public}]) {
  id: ID!
  name: String!
  parent: BelongsToParentModel! @hasOne
}

type BelongsToParentModel @model @auth(rules: [{allow: public}]) {
  id: ID!
  name: String!
}
```

Another option would be to remove all use of "parent" and "child" from the schema, but this would be a much larger change.

Reply:
As GraphQL transformer v1 is still being supported as legacy feature (not recommended for new projects), I think we can leave the original integ tests suite as is. When adding the tests suite for v2, we can use better naming. Thoughts?

Reply:
How long is v1 being supported? I think it is fine to consider it out of scope for this PR, but if we will be supporting v1 for a while it might be nice to update the v1 schema at some point too.

Reply:
I'm not aware of a clear deprecation plan. We can improve while implementing the test suites for v2.

Reply:
Here's is what I used for testing hasOne/belongsTo (agree that current naming is suuuper confusing)

**v1**

```graphql
type ParentModel @model {
  id: ID!
  name: String!
  childID: ID
  child: ChildModel @connection(fields: ["childID"]) @hasOne
}

type ChildModel @model {
  id: ID!
  name: String!
  parent: ParentModel! @connection # @belongsTo
}
```

**v2**

```graphql
type ParentModel @model {
  id: ID!
  name: String!
  childID: ID
  child: ChildModel @hasOne(fields: ["childID"])
}

type ChildModel @model {
  id: ID!
  name: String!
  parent: ParentModel! @belongsTo
}
```